### PR TITLE
Create new command 

### PR DESCRIPTION
**Pull request by Ludea**
New command to create derivative.
Currently support only opensuse

### CODE REVIEW

**Comment by mudler** [comment]
That's a really nice touch! This was part of https://github.com/rancher-sandbox/cOS-toolkit/issues/108 so it definitely makes sense to have it

File: cmd/new.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+		if err != nil {
+			cfg.Logger.Errorf("Error reading config: %s\n", err)
+		}
+
+		flavor := args[0]
+
+		client := &getter.Client{
+			Ctx: context.Background(),
+			Dst: "derivatives/opensuse",
```
Comment (mudler) [inline cmd/new.go]:
```suggestion
			Dst: fmt.Sprintf("derivatives/%s",flavor),
```

File: cmd/new.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			Dir: true,
+			Src: "github.com/rancher-sandbox/cOS-toolkit//examples/standard",
```
Comment (mudler) [inline cmd/new.go]:
```suggestion
			Src: "github.com/rancher-sandbox/cOS-toolkit/examples/standard",
```

File: cmd/new.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			Mode: getter.ClientModeDir,
+			Detectors: []getter.Detector{
+				&getter.GitHubDetector{},
+			},
+		}
+
+		var destination string
```
Comment (mudler) [inline cmd/new.go]: I guess we can move this up and generate it for each flavor

File: cmd/new.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		switch flavor {
+		case "opensuse": destination = "derivatives/opensuse"
+		case "fedora":
+		case "ubuntu":
+		default: cfg.Logger.Errorf("Unsupported flavor")
+			return errors.New("Unsupported flavor")
+		}
+
+		client = &getter.Client{
```
Comment (mudler) [inline cmd/new.go]: Looks like a dup here?

File: cmd/new.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	SilenceUsage:  true, // Do not show usage on error
+	SilenceErrors: true, // Do not propagate errors down the line, we control them
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg, err := config.ReadConfigRun(viper.GetString("config-dir"), &mount.FakeMounter{})
+
+		if err != nil {
+			cfg.Logger.Errorf("Error reading config: %s\n", err)
+		}
+
+		flavor := args[0]
```
Comment (mudler) [inline cmd/new.go:40]: I'd suggest to rather gate here the ones currently supported by the examples

File: cmd/new.go (hunk 2 above)
Comment (Ludea) [reply cmd/new.go]: https://github.com/hashicorp/go-getter#subdirectories
I need the double / to download the subfolder

File: cmd/new.go (hunk 1 above)
Comment (Ludea) [reply cmd/new.go]: Done

File: cmd/new.go (hunk 4 above)
Comment (Ludea) [reply cmd/new.go]: Done

File: cmd/new.go (hunk 5 above)
Comment (Ludea) [reply cmd/new.go:40]: Will do separate PR to add more examples

**Review by mudler — COMMENTED** [review]
Looks good here, any thoughts @Itxaka @davidcassany ?

**Review by davidcassany — APPROVED** [review]
Looks good to me, nice!

**Comment by mudler** [comment]
@Ludea we need commits to be signed-off, otherwise looks good to merge!

**Pull request merged**